File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = [arr[0]]
    # [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    for i in range(1,len(arr)):
        if answer[-1]==arr[i]:
            continue
        answer.append(arr[i])
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public static Object[] solution(int []arr) {
//        int[] answer = {};
//        ArrayList<Integer> answer=new ArrayList<Integer>();
        Stack<Integer> stack = new Stack<>();
        stack.push(arr[0]);
        int n=arr.length;

        for(int i=1;i<n;i++){
            if(stack.peek()==arr[i]){
                continue;
            }
            else {
                stack.push(arr[i]);
            }
        }

        Object[] answer=stack.toArray();


        return answer;
    }
}